x-pack/platform/plugins/shared/lens/public/app_plugin/app.tsx

```diff
@@ -355,10 +348,10 @@ export function App({
 
   // keeping the initial doc state created by the context
   useEffect(() => {
-    if (lastKnownDoc && !initialDocFromContext) {
-      setInitialDocFromContext(lastKnownDoc);
+    if (currentDoc && !initialDocFromContext) {
+      setInitialDocFromContext(currentDoc);
     }
-  }, [lastKnownDoc, initialDocFromContext]);
+  }, [currentDoc, initialDocFromContext]);
```

Comment:
The scope of this is to keep initial current doc around. To avoid a re-render caused by the `setInitialDocFromContext` we should adopt a different strategy (that's looks a bit uglier but works the same way:
```
const initialDocFromContext = React.useRef(undefined)
  if (!initialDocFromContext.current && currentDoc) {
    initialDocFromContext.current = currentDoc
  }
  ```
  In this way you should avoid the re-rendered caused by that useEffect